app/javascript/packages/rails-i18n-webpack-plugin/rails-i18n-webpack-plugin.js

```diff
@@ -0,0 +1,259 @@
+const fs = require('fs/promises');
+const { format } = require('util');
+const path = require('path');
+const YAML = require('yaml');
+const ExtractKeysWebpackPlugin = require('./extract-keys-webpack-plugin.js');
+
+/**
+ * @typedef RailsI18nWebpackPluginOptions
+ *
+ * @prop {string} template String to format using locale data object, generated using util.format.
+ * @prop {string} configPath Root path for locale configuration data.
+ * @prop {string} defaultLocale Default locale to use if string data is missing for desired locale.
+ * @prop {string} missingStringValue Value to use if string data is missing for both desired and
+ * default locales.
+ */
+
+/**
+ * Returns the value in the object at the given key path.
+ *
+ * @example
+ * ```js
+ * const value = dig({ a: { b: { c: 'foo' } } }, ['a', 'b', 'c']);
+ * // 'foo'
+ * ```
+ *
+ * @param {Record<string, any>} object
+ * @param {string[]} keyPath
+ *
+ * @return {any}
+ */
+function dig(object, keyPath) {
+  let result = object;
+  for (const segment of keyPath) {
+    result = result?.[segment];
```

Comment:
just a note from our discussion in Slack --- ran into syntax issues on this line with node 12, because the null-safe navigator is too new for us

Reply:
Follow-up at #5227